k8s: Rename CEPs to match their pod name

raybejjani opened the pull request:
User feedback suggests it's easier to correlate a CEP with it's pod if it has a similar name. The name may as well be the same, since one fetches one type or the other explicitly.

```
$ kubectl get pods
NAME                    READY     STATUS    RESTARTS   AGE
app1-799c454b56-6rxtz   1/1       Running   0          4d
app1-799c454b56-v69c9   1/1       Running   0          4d
app2                    1/1       Running   0          4d
app3                    1/1       Running   0          4d
$ kubectl get cep
NAME                    AGE
app1-799c454b56-6rxtz   2m
app1-799c454b56-v69c9   2m
app2                    2m
app3                    2m
```

I also threw in a small fixup of the k8s cilium client getter function but I can make that it's own PR.

raybejjani committed:
k8s: Rename CEPs to match their pod name

User feedback suggests it's easier to correlate a CEP with it's pod if
it has a similar name. The name may as well be the same, since one
fetches one type or the other explicitly.

Signed-off-by: Ray Bejjani <ray@covalent.io>

eloycoto commented on pkg/endpoint/endpoint.go:
```diff
@@ -266,37 +266,42 @@ func RunK8sCiliumEndpointSyncGC() {
 
+				clusterPods, err := k8sClient.CoreV1().Pods("").List(meta_v1.ListOptions{})
+				if err != nil {
+					scopedLog.WithError(err).Error("Unable to get list of pods to GC CEP")
```
Shouldn't here return an error?

raybejjani replied:
yes. I remember making this a log but I have no idea why!

raybejjani committed:
k8s: Guard against failed cilium client setup

We ensure a k8s cilium client is created with a Once.Do but in the case
that we encounter an error in the Once, other threads blocked on it or
later calls to this function will dereference a nil in the return.

Signed-off-by: Ray Bejjani <ray@covalent.io>

tgraf commented on pkg/endpoint/endpoint.go:
```diff
@@ -220,6 +220,13 @@ func getCiliumClient() (ciliumClient cilium_client_v2.CiliumV2Interface, err err
+	// and we are a different thread of execution (or a later call of this
+	// function)
+	if ciliumEndpointSyncControllerK8sClient == nil {
+		return nil, errors.New("No initialised k8s Cilium CRD client")
```
Grammar

tgraf commented on pkg/endpoint/endpoint.go:
```diff
@@ -273,37 +273,43 @@ func RunK8sCiliumEndpointSyncGC() {
-					ownerNode := parts[0]
-
-					if _, found := nodes[node.Identity{Name: ownerNode}]; !found {
+					if _, found := clusterPodSet[cep.Name]; !found {
```
namespace match
